### PR TITLE
post: ng and cli v16

### PR DESCRIPTION
**Pull request by cexbrayat**

### CODE REVIEW

File: _posts/2023-05-03-angular-cli-16.0.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    ng new my-app --standalone
+
+This is great to start new projects 😍
+
+now that the option is available, we switched all the exercises of our 
+[Angular training](angular-exercises.ninja-squad.com/) to use a standalone application
+with standalone components.
+
+We also re-wrote our ebook entirely to now introduce standalone concepts from the beginning,
+and add the concept of modules only matter, when it makes sense!
```
Comment (jnizet) [inline _posts/2023-05-03-angular-cli-16.0.md:31]: > and add the concept of modules only matter, when it makes sense!

I don't understand precisely what you meant here. Needs to be rephrased.

File: _posts/2023-05-03-angular-cli-16.0.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+The support is far from being complete, but it is promising.
+It uses the new esbuild builder under the hood,
+and the support should improve in the next releases.
+
+Jest does not run in a browser and uses JSDOM instead.
+That means that you don't have the same experience that you can have with Karma,
+but as Karma is now deprecated, an alternative is welcome.
+
+The Angular team also announced that they are working on the support of
+[Web Test Runner](https://modern-web.dev/docs/test-runner/overview/),
```
Comment (jnizet) [inline _posts/2023-05-03-angular-cli-16.0.md:77]: Unrelated to the review, but I'm looking forward to this much more than Jest.

File: _posts/2023-05-03-what-is-new-angular-16.0.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+Angular v16 added the possibility to mark an input as required,
+with `@Input({ required: true})`:
+
+    @Input({ required: true }) user!: UserModel;
+
+In that case, if the parent component does not pass the input, 
+then the compiler will throw an error.
+
+This has been a long-awaited feature, we're happy to this land in Angular!
```
Comment (jnizet) [inline _posts/2023-05-03-what-is-new-angular-16.0.md:46]: to **see** this land

File: _posts/2023-05-03-what-is-new-angular-16.0.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+The behavior is enabled by default if you use `provideClientHydration()`,
+but can be disabled with `provideClientHydration(withNoHttpTransferCache())`.
+You can also disable the DOM reuse with `withNoDomReuse()`.
+
+Note that this is a developer preview, and the API may change in the future.
+There are also a few pitfalls to be aware of.
+For example, the HTML must be valid when generated on the server (whereas the browser is more forgiving).
+The DOM must also be the same on the server and the client,
+so you can't manipulate the server-rendered DOM before sending it to the client.
+If some parts of your templates will not produce the same result on the server and the client, you can skip them by adding `ngSkipHydration` to the element or component.
```
Comment (jnizet) [inline _posts/2023-05-03-what-is-new-angular-16.0.md:82]: will not -> don't

File: _posts/2023-05-03-what-is-new-angular-16.0.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+i18n is also not yet supported, but that should come soon.
```
Comment (jnizet) [inline _posts/2023-05-03-what-is-new-angular-16.0.md:83]: not supported yet

File: _posts/2023-05-03-what-is-new-angular-16.0.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+There is no rush to use the new decorators instead of the "legacy" ones, but it's something to keep in mind as I wouldn't be surprised if we have to migrate away from `experimentalDecorators` in the future
+
+## Styles removal opt-in
+
+Angular v16 introduces a new opt-in feature to remove the styles of a component
+when its last instance is destroyed.
+
+This will be the default behavior in the future, but you can already opt in with:
+
+    { provide: REMOVE_STYLES_ON_COMPONENT_DESTROY, useValue: true }
```
Comment (jnizet) [inline _posts/2023-05-03-what-is-new-angular-16.0.md:144]: I didn't even know this wasn't the case already. So they add them to the styles on first use, and leave them there forever?

File: _posts/2023-05-03-what-is-new-angular-16.0.md [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+
+You can now directly use `RouterModule.forRoot([])` or `providerRouter([])` in your tests.
+
+Last but not least, the router now offers the possibility to bind parameters as inputs.
+
+To do so, you need to configure the router with `withComponentInputBinding`:
+
+    provideRouter(routes, withComponentInputBinding())
+
+With this option, a component can declare an input with the same name as a route parameter or data, and Angular will automatically bind the value of the parameter or data to this input.
```
Comment (jnizet) [inline _posts/2023-05-03-what-is-new-angular-16.0.md:174]: Might be nice to say if this also supports query parameters.

**Review by jnizet — CHANGES_REQUESTED** [review]

File: _posts/2023-05-03-angular-cli-16.0.md (hunk 2 above)
Comment (cexbrayat) [reply _posts/2023-05-03-angular-cli-16.0.md:77]: Hihi I suspected so 😉

File: _posts/2023-05-03-what-is-new-angular-16.0.md (hunk 6 above)
Comment (cexbrayat) [reply _posts/2023-05-03-what-is-new-angular-16.0.md:144]: Yep

**Pull request merged**